### DISCUSSION TITLE
Questions on config

### DISCUSSION THREAD

**Discussion by micaelarealign** [brianpetro/obsidian-smart-connections#902] [Q&A]
First off, thanks for the update. 

But I have some questions since upgrading from v1. I thought asking these questions could help the community understand certain settings that may not be that clear to the average user such as myself. Although I am quite familiar with how AI works, some of these things still escape me in this plugin. Maybe your answers could be added as tooltips in the settings.

1. Minimum length of entity to embed.

It's set to default 300. Does that mean notes less than 300 characters won't be included in Smart Connections or Smart Chat? What exactly does it mean that they won't be embedded? What's the implication? What if I set it to 1? or 0?

2. Embed blocks

Enabled by default... what happens if you disable it? What does this actually do? Does it still embed the note but not the blocks? What benefit does embedding blocks offer?

3. Legacy transformers

What does V3 offer over V2? I'm running on a laptop with integrated graphics. What effect does this have?

4. Embedding model platform

Transformers vs OpenAI - what is the difference?

I'm sure there are more questions to ask but these are the basic ones that have me wondering.

**Comment by brianpetro** [comment]
_Note: moved to discussions so the issues section can be reserved for clear bugs and feature requests_

1. it's length in characters. Decreasing increases the number of embeddings. I wouldn't decrease this unless you are aware of the consequences (mostly performance related.)
2. blocks are sub-sections in notes. Enabling increases the number of embeddings. It should be turned on if you have many long notes.
3. v2 (legacy) does not use the GPU. Sometimes v3 doesn't work, so using legacy is required.
4. Transformers are local embedding models. It was mentioned in the [release notes](https://github.com/brianpetro/obsidian-smart-connections/releases/tag/2.3.37) and will also be clarified in the UI later

**Comment by micaelarealign** [comment]
I appreciate the answers but I am still a little bit in the dark...

1. what exactly are the consequences? I'm not that concerned with performance, so what would decreasing it offer in terms of benefits?

3. I understand that v2 does not use GPU (it is stated in the settings, after all), but my question was what is the difference?
